relay/README.md

```diff
@@ -1,69 +1,48 @@
+> Circuit Switching for libp2p, also known as TURN or Relay in Networking literature.
 
-Implementations
-- [js-libp2p-circuit](https://github.com/libp2p/js-libp2p-circuit) -- status: started
-- [go-libp2p-circuit](https://github.com/libp2p/go-libp2p-circuit) -- status: ready-to-start
+## Implementations
+
+- [js-libp2p-circuit](https://github.com/libp2p/js-libp2p-circuit)
+- [go-libp2p-circuit](https://github.com/libp2p/go-libp2p-circuit)
+
+## Table of Contents
 
-Table of Contents
 - [Overview](#overview)
 - [Dramatization](#dramatization)
 - [Addressing](#addressing)
 - [Wire protocol](#wire-protocol)
 - [Interfaces](#interfaces)
+- [Implementation Details](#implementation-details)
 - [Removing existing relay protocol](#removing-existing-relay-protocol)
 
-Note: this document is currently in the process of being double-checked and updated. Please see [libp2p/specs#15 (comment)](https://github.com/libp2p/specs/pull/15#issuecomment-299949023) before using it as the basis for an implementation.
-
-
 ## Overview
 
-The circuit relay is a means of establishing connectivity between
-libp2p nodes (such as IPFS) that wouldn't otherwise be able to connect to each other.
+The circuit relay is a mean to establish connectivity between libp2p nodes (e.g IPFS) that wouldn't otherwise be able to establish a direct connection to each other.
 
-This helps in situations where nodes are behind NAT or reverse proxies,
-or simply don't support the same transports (e.g. go-ipfs vs. browser-ipfs).
-libp2p already has modules for NAT ([go-libp2p-nat](https://github.com/libp2p/go-libp2p-nat)),
-but these don't always do the job, just because NAT traversal is complicated.
-That's why it's useful to have a simple relay protocol.
+Relay is needed in situations where nodes are behind NAT, reverse proxies, firewalls and/or simply don't support the same transports (e.g. go-ipfs vs. browser-ipfs). Even though libp2p has modules for NAT traversal ([go-libp2p-nat](https://github.com/libp2p/go-libp2p-nat)), piercing through NATs isn't always an option. The circuit relay protocol exists to overcome those scenarios.
 
-Unlike a transparent **tunnel**, where a libp2p peer would just proxy a
-communication stream to a destination (the destination being unaware of the
-original source), a circuit-relay makes the destination aware of the original
-source and the circuit followed to establish communication between the two.
-This provides the destination side with full knowledge of the circuit which,
-if needed, could be rebuilt in the opposite direction.
+Unlike a transparent **tunnel**, where a libp2p peer would just proxy a communication stream to a destination (the destination being unaware of the original source), a circuit relay makes the destination aware of the original source and the circuit followed to establish communication between the two. This provides the destination side with full knowledge of the circuit which, if needed, could be rebuilt in the opposite direction.
```

Comment:
Do we really need this? It seems like there could be a lot of complexity involved in making this work properly, as opposed to leaving the job of re-establishing a dropped connection to the `source`. So far I have left this out in the js implementation.